src/server/req_stat.c

```diff
@@ -445,77 +442,23 @@ req_stat_que(struct batch_request *preq)
 	preply->brp_choice = BATCH_REPLY_CHOICE_Status;
 	CLEAR_HEAD(preply->brp_un.brp_status);
 
-	memset(&attr, 0, sizeof(attr));
-	if (preq->rq_extend != NULL) {
-		rc = sched_attr_def[SCHED_ATR_partition].at_decode(&attr, ATTR_partition, NULL, preq->rq_extend);
-		if (rc) {
-			(void) reply_free(preply);
-			req_reject(rc, bad, preq);
-			return;
-		}
-	}
-
 	if (type == 0) {	/* get status of the one named queue */
-		if (preq->rq_extend != NULL) {
-			if (pque->qu_attr[QA_ATR_partition].at_flags & ATR_VFLAG_SET) {
-				/* search for the partition in the attr */
-				for (k = 0; k < attr.at_val.at_arst->as_usedptr; k++) {
-					if ((attr.at_val.at_arst->as_string[k] != NULL) &&
-						(!strcmp(pque->qu_attr[QA_ATR_partition].at_val.at_str, attr.at_val.at_arst->as_string[k]))) {
-						consider_q = 1;
-						break;
-					}
-				}
-			}
-		} else {
-			if (preq->rq_conn == dflt_scheduler->scheduler_sock) {
-				if (!(pque->qu_attr[QA_ATR_partition].at_flags & ATR_VFLAG_SET))
-					consider_q = 1;
-			} else {
-				consider_q = 1;
-			}
-		}
-		if (consider_q) {
-			rc = status_que(pque, preq, &preply->brp_un.brp_status);
-			consider_q = 0;
-		}
+		rc = status_que(pque, preq, &preply->brp_un.brp_status);
+
 	} else {	/* get status of queues */
-		for (pque = (pbs_queue *)GET_NEXT(svr_queues); pque;pque = (pbs_queue *)GET_NEXT(pque->qu_link)) {
-			if (preq->rq_extend != NULL) {
-				if (pque->qu_attr[QA_ATR_partition].at_flags & ATR_VFLAG_SET) {
-					/* search for the partition in the attr */
-					for (k = 0; k < attr.at_val.at_arst->as_usedptr; k++) {
-						if ((attr.at_val.at_arst->as_string[k] != NULL) &&
-							(!strcmp(pque->qu_attr[QA_ATR_partition].at_val.at_str, attr.at_val.at_arst->as_string[k]))) {
-							consider_q = 1;
-							break;
-						}
-					}
-				}
-				if (!consider_q) {
-					continue;
-				}
-			} else {
-				if (preq->rq_conn == dflt_scheduler->scheduler_sock) {
-					if (!(pque->qu_attr[QA_ATR_partition].at_flags & ATR_VFLAG_SET))
-						consider_q = 1;
-				} else {
-					consider_q = 1;
-				}
-			}
-			if (consider_q) {
-				rc = status_que(pque, preq, &preply->brp_un.brp_status);
-				if (rc != 0) {
-					if (rc == PBSE_PERM)
-						rc = 0;
-					else
-						break;
-				}
-				consider_q = 0;
+
+		pque = (pbs_queue *)GET_NEXT(svr_queues);
+		while (pque) {
+			rc = status_que(pque, preq, &preply->brp_un.brp_status);
+			if (rc != 0) {
+				if (rc == PBSE_PERM)
+					rc = 0;
```

Comment:
Why do you check for rc == PBSE_PERM here, but not on the other side of the if()?

Reply:
Now that we have moved the filtering of queues/nodes to scheduler, I did not touch anything in this part of the code.